cibuildwheel/macos.py

```diff
@@ -263,29 +263,47 @@ def setup_python(
         )
         sys.exit(1)
 
-    # Set MACOSX_DEPLOYMENT_TARGET to 10.9, if the user didn't set it.
+    # deployment target:
+    # retrieve the one supplied by the user, otherwise 10.9 is supported with every installer.
+    # An exception can be made for wheel built explicitly for arm64, this will be fixed later on.
     # PyPy defaults to 10.7, causing inconsistencies if it's left unset.
-    env.setdefault("MACOSX_DEPLOYMENT_TARGET", "10.9")
+    # it's valid to set e.g. MACOSX_DEPLOYMENT_TARGET=10.12.1 or MACOSX_DEPLOYMENT_TARGET=11
+    # always normalize as tuple(major, minor)
+    deployment_target = tuple(map(int, env.get("MACOSX_DEPLOYMENT_TARGET", "10.9").split(".")[:2]))
+    if deployment_target[0] >= 11:
+        # starting with macOS 11, the minor version in fact represents patch version
+        deployment_target = (deployment_target[0], 0)
+    if len(deployment_target) < 2:
+        # even though it didn't make much sense before macOS 11, it was valid to set
+        # MACOSX_DEPLOYMENT_TARGET=10 which is 10.0
+        deployment_target = (deployment_target[0], 0)
```

Comment:
I tripped up on this a little, perhaps something like the following would be easier to read?

```suggestion
    parts = env.get("MACOSX_DEPLOYMENT_TARGET", "10.9").split(".")
    if len(parts) < 2:
        deployment_target = int(parts[0]), 0
    else:
        deployment_target = int(parts[0]), int(parts[1])

    if deployment_target[0] >= 11:
        # starting with macOS 11, we retain compatibility across minor versions
        deployment_target = (deployment_target[0], 0)
```